File: bench/tasks/13-temperatures/zig/main.zig
const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    const temps = [_]i64{ 22, 18, 25, 31, 29, 17, 24, 28, 20, 26 };
    var total: i64 = 0;
    var hot: i64 = 0;
    var mild: i64 = 0;
    var cold: i64 = 0;
    for (temps) |t| {
        total += t;
        if (t >= 28) {
            hot += 1;
        } else if (t >= 20) {
            mild += 1;
        } else {
            cold += 1;
        }
    }
    const avg = total / @as(i64, @intCast(temps.len));
    try stdout.print("Days measured: {}\n", .{temps.len});
    try stdout.print("Average: {}\n", .{avg});
    try stdout.print("Hot days (28+): {}\n", .{hot});
    try stdout.print("Mild days: {}\n", .{mild});
    try stdout.print("Cold days (<20): {}\n", .{cold});
}

File: bench/tasks/13-temperatures/rust/main.rs
fn main() {
    let temps = [22, 18, 25, 31, 29, 17, 24, 28, 20, 26];
    let total: i32 = temps.iter().sum();
    let avg = total / temps.len() as i32;
    let hot = temps.iter().filter(|&&t| t >= 28).count();
    let mild = temps.iter().filter(|&&t| t >= 20 && t < 28).count();
    let cold = temps.iter().filter(|&&t| t < 20).count();
    println!("Days measured: {}", temps.len());
    println!("Average: {}", avg);
    println!("Hot days (28+): {}", hot);
    println!("Mild days: {}", mild);
    println!("Cold days (<20): {}", cold);
}
